### ISSUE TITLE
implement adversarial example generation

### ISSUE THREAD

**Issue by tiffanynle** [tiffanynle/advML-atk-n-def#2]
implement methods to generate adversarial examples using PyTorch:
- fast gradient sign method (FGSM)
- projected gradient descent (PGD)
- ...

relevant papers:
https://arxiv.org/abs/1412.6572
https://arxiv.org/abs/1706.06083
